docs/08_getting_started.md

```diff
@@ -0,0 +1,329 @@
+we recommend using a newer OkHttp version, because the version used in the example will only receive security updates until December 2021.
+
+- [Downloader implementation for NewPipe Extractor tests](https://github.com/TeamNewPipe/NewPipeExtractor/blob/master/extractor/src/test/java/org/schabi/newpipe/downloader/DownloaderTestImpl.java)
+- [Downloader implementation of the NewPipe app](https://github.com/TeamNewPipe/NewPipe/blob/master/app/src/main/java/org/schabi/newpipe/DownloaderImpl.java)  
+Note: This implementation contains some additional methods which are not necessary for general use cases.
+
+
+Depending on your use case and your downloader implementation, we suggest you to make use of the singleton pattern.
+
+#### Browser Agent
+
+When making requests to websites, you should define a user agent for the request headers. We have build our tests on
+```
+Mozilla/5.0 (Windows NT 10.0; rv:78.0) Gecko/20100101 Firefox/78.0
+```
+For this reason, we strongly recommend using that agent or any other that is used in the [testimplementation in NewPipe Extractor](https://github.com/TeamNewPipe/NewPipeExtractor/blob/master/extractor/src/test/java/org/schabi/newpipe/downloader/DownloaderTestImpl.java).  
+However, a service might want to override the default user agent.
+This is sometimes necessary to get special versions of a website (e.g. mobile or legacy version).
+Please respect that when implementing the `execute(@Nonnull Request request)` method.
+
+## Initialize NewPipe
+
+``` Java
+NewPipe.init(
+    /* 
+      New instance of your downloader implementation.
+      If you implemented the singleton pattern, use DownlaoderImpl.getInstance() instead.
+    */
+    new DownloaderImpl()
+);
+```
+You can also set a default localization 
+``` Java
+NewPipe.init(
+    new DownloaderImpl(),
+    new Localization("EN", "US")
+);
+```
+and a default content country. However, not all services support those, yet.
+``` Java
+NewPipe.init(
+    new DownloaderImpl(),
+    new Localization("EN", "US"),
+    new ContentCountry("EN-US")
+);
+```
+
+
+
+## Using Service
+
+We can any service that NewPipe supports from `ServiceList`.
+
+```java
+YoutubeService youtube = ServiceList.YouTube;
+// SoundcloudService soundCloud = ServiceList.SoundCloud;
+// ...
+```
+
+### Some examples for **youtube** service:
+
+
+- Suggestions
+
+```java
+youtube.getSuggestionExtractor().suggestionList("Hello");
+```
+
+- Search results
+
+```java
+SearchExtractor page = youtube.getSearchExtractor("Hello");
+page.fetchPage();
+
+// results list
+page.getInitialPage().getItems();
+```
+
+
+- get video stream links:
+
+```java
+StreamExtractor page = youtube.getStreamExtractor("https://www.youtube.com/watch?v=1YGCQRTSOSI");
+page.fetchPage();
+
+// list of streams
+page.getVideoStreams();
+```
+
+## Examples:
+- [Android example](https://github.com/cloneconf/newpipe-android-example) by @cloneconf.
+- [Java example](https://github.com/cloneconf/newpipe-java-example) by @cloneconf.
```

Comment:
I'm not a fan of linking these repositories. I did not and won't look through them to evaluate whether they are good examples.

I find the unit tests in the extractor should be a decent place to get some examples. They are also maintained by us.